go/vt/vtgate/semantics/analyzer.go

```diff
@@ -290,35 +289,36 @@ func (a *analyzer) checkForInvalidConstructs(cursor *sqlparser.Cursor) error {
 			return nil
 		}
 		if a.scoper.currentScope().parent != nil {
-			return NewError(CantUseOptionHere, errMsg)
+			return &CantUseOptionHereError{Msg: errMsg}
 		}
 	case *sqlparser.Nextval:
 		currScope := a.scoper.currentScope()
 		if currScope.parent != nil {
-			return NewError(CantUseOptionHere, "Incorrect usage/placement of 'INTO'")
+			// TODO: untested
+			return &CantUseOptionHereError{Msg: "Incorrect usage/placement of 'INTO'"}
 		}
 		if len(currScope.tables) != 1 {
-			return NewError(NextWithMultipleTables)
+			return &NextWithMultipleTablesError{CountTables: len(currScope.tables)}
 		}
 		vindexTbl := currScope.tables[0].GetVindexTable()
 		if vindexTbl == nil {
-			return NewError(MissingInVSchema)
+			return &MissingInVSchemaError{}
 		}
 		if vindexTbl.Type != vindexes.TypeSequence {
-			return NewError(NotSequenceTable)
+			return &NotSequenceTableError{Table: vindexTbl.Name.String()}
 		}
 	case *sqlparser.JoinTableExpr:
 		if node.Join == sqlparser.NaturalJoinType || node.Join == sqlparser.NaturalRightJoinType || node.Join == sqlparser.NaturalLeftJoinType {
-			return NewError(UnsupportedNaturalJoin, node.Join.ToString())
+			return &UnsupportedNaturalJoinError{Join: node.Join.ToString()}
```

Comment:
I think it's useful here to have the actual sqlparser node? A user can always create a string out of that, but it can be really useful to have the actual tree node for other analysis?

Reply:
done